File: DemoClient/projects/ng-realtime-database/src/lib/models/prefilter/take-prefilter.ts
import {IPrefilter} from './iprefilter';

export class TakePrefilter<T> implements IPrefilter<T> {
  prefilterType = 'TakePrefilter';
  number: number;

  constructor(number: number) {
    this.number = number;
  }

  execute(values: T[]) {
    return values;
  }

  public hash() {
    return `${this.prefilterType},${this.number}`;
  }
}

File: SapphireDb/Internal/Prefilter/TakePrefilter.cs
using System;
using System.Linq;

namespace SapphireDb.Internal.Prefilter
{
    public class TakePrefilter : IPrefilter
    {
        public int Number { get; set; }

        public IQueryable<object> Execute(IQueryable<object> array)
        {
            return array.Take(Number);
        }

        public void Initialize(Type modelType, IServiceProvider serviceProvider) { }

        public void Dispose()
        {
            
        }
        
        public string Hash()
        {
            return $"TakePrefilter,{Number}";
        }
    }
}
